### PR TITLE
updating Centos to use C++14 ready compiler

### PR DESCRIPTION
**Pull request by roigcarlo**
**📝 Description**
Updating centos docker

**🆕 Changelog**
- Centos docker now uses C++14 compiler and python 3.8.10

### CODE REVIEW

**Comment by roigcarlo** [comment]
@pooyan-dadvand This should be ready, the new images will only be pushed when we merge this. So I had to disable the centos CI temporaly (deadlock).

Once we have it working I will re enable again

File: scripts/docker_files/docker_file_ci_centos_7/DockerFile [hunk 1]
```diff
@@ -4,28 +4,43 @@ USER root
...
-RUN tar xzf Python-3.5.4.tgz -C /usr/src
-RUN ls /usr/src/Python-3.5.4
-RUN sh /usr/src/Python-3.5.4/configure --enable-optimizations --with-lto --enable-shared LDFLAGS="-Wl,-rpath /usr/local/lib"
+RUN wget https://www.python.org/ftp/python/3.8.10/Python-3.8.10.tgz
+RUN tar xzf Python-3.8.10.tgz -C /usr/src
+RUN ls /usr/src/Python-3.8.10
+RUN sh /usr/src/Python-3.8.10/configure --enable-optimizations --with-lto --enable-shared LDFLAGS="-Wl,-rpath /usr/local/lib"
 RUN make altinstall
-RUN rm Python-3.5.4.tgz
-RUN python3.5 -m pip install 'cx_Freeze==6.1' numpy sympy
```
Comment (ddiezrod) [inline scripts/docker_files/docker_file_ci_centos_7/DockerFile]: you are not installing numpy and sympy? I think we use them in some places in the code

Comment (roigcarlo) [reply scripts/docker_files/docker_file_ci_centos_7/DockerFile]: Oh good catch, I forgot to re-add them for 3.8

**Review by pooyan-dadvand — APPROVED** [review]
Thank you! 

Let's merge it so we can make the centos test required again

**Comment by rubenzorrilla** [comment]
Does this mean that we can start using c++14 features?

**Comment by roigcarlo** [comment]
It means Centos Image should support them now (actually it should support C++17)

**Pull request merged**